src/swap/components/SwapProvider.tsx

```diff
@@ -68,6 +69,12 @@ export function SwapProvider({
   // Refreshes balances and inputs post-swap
   const resetInputs = useResetInputs({ from, to });
 
+  const getMaxSlippage = useCallback(() => {
+    if (lifeCycleStatus.statusData && lifeCycleStatus.statusName !== 'error') {
+      return lifeCycleStatus.statusData.maxSlippage;
+    }
+    return experimental.maxSlippage || DEFAULT_MAX_SLIPPAGE;
+  }, [lifeCycleStatus, experimental.maxSlippage]);
```

Comment:
I feel like there is something logically wrong here.... I am going to read this more carefully.

Reply:
I feel like, the fact that `statusData` it's not the same when an `error` occurs, creates this weird odd logic, which will have more side effects.

Like, if an error occurs, we lose whatever was the last `maxSliappage` a user might have customized.

Plus, adding those weird if/else, creates a precedent where for other options we need to play this odd if/else

Reply:
Let me open a PR to help write down my thinking.

Reply:
https://github.com/coinbase/onchainkit/pull/1234